### ISSUE TITLE
Registry v2 wallet Error SSL_PROTOCOL_ERR

### ISSUE THREAD

**Issue by davidthomasparks** [WalletConnect/walletconnect-monorepo#950]
Hey there,

As of 4 days ago we've observed mobile links no longer appearing on our Wallet connect modals. I think we can potentially trace this to the depreciation of the old mobile registry. On our production environment we are getting the following error when opening the WC modal:

`GET https://registry.walletconnect.com/api/v2/wallets net::ERR_SSL_PROTOCOL_ERROR`
<img width="590" alt="Screen Shot 2022-04-14 at 12 37 06 PM" src="https://user-images.githubusercontent.com/97011546/163463527-c630aca4-514b-4387-83b2-84ed5024410b.png">

Can someone shed some light on a temporary fix here? Our deps are as follows: 

```
    "@walletconnect/ethereum-provider": "^1.7.5",
    "@web3-react/coinbase-wallet": "8.0.23-beta.0",
    "@web3-react/core": "^8.0.23-beta.0",
    "@web3-react/eip1193": "8.0.14-beta.0",
    "@web3-react/empty": "8.0.10-beta.0",
    "@web3-react/metamask": "8.0.16-beta.0",
    "@web3-react/network": "8.0.15-beta.0",
    "@web3-react/types": "8.0.10-beta.0",
    "@web3-react/url": "8.0.14-beta.0",
    "@web3-react/walletconnect": "^8.0.26-beta.0",
```

**Comment by imthatcarlos** [comment]
+1 literally was dealing with this the past 2 hours. I tried setting the legacy api manually for `qrcodeModalOptions.registryUrl` and ran into some parsing error, so figure it wasn't valid in the latest version of the sdk, tried deprecating, same parsing error. Guess we're waiting on this v2 api to not have this SSL error ?

**Comment by imthatcarlos** [comment]
related https://github.com/WalletConnect/walletconnect-monorepo/issues/933

**Comment by davidthomasparks** [comment]
Tried your registry fix too @imthatcarlos and am getting the same parsing error. Thanks for the related issue. Hoping they can resolve this soon, followed up in the WC discord hoping to get some eyes on this blocking problem.

**Comment by imthatcarlos** [comment]
@davidthomasparks it's probably too much of a hack/workaround but i noticed that there was a slight difference in the response documented and what the legacy api responds with, so maybe creating an intermediate api that just takes the legacy response, formats it for this: https://docs.walletconnect.com/2.0/api/registry-api#get-apiv1wallets and the setting your intermeidate api as the registry. again, too hacky but if you need it in prod..

**Comment by davidthomasparks** [comment]
Now this is the kinda hack I like @imthatcarlos haha. I will give this a shot. We do in fact need it in prod. Thanks for the idea

**Comment by imthatcarlos** [comment]
haha nice yeah no problem! good luck, and lmk if it works 🙏 we're not in prod yet, but would be nice to have a fallback option in the interim. cheers

**Comment by davidthomasparks** [comment]
Wow it worked @imthatcarlos thanks legend lol. We're using Next so here was the API route:

```
import { withSentry } from "@sentry/nextjs";

const handler = async (req, res) => {
    const { method } = req;

    res.setHeader(
        'Cache-Control',
        'public, s-maxage=59, stale-while-revalidate=600'
    )

    switch (method) {
        case "GET":
            try {
                const wallets = await fetch(`https://registry.walletconnect.org/data/wallets.json`)
                const data = await wallets.json();
                let listings = [];
                Object.keys(data).forEach((wallet) => {
                    listings.push(data[wallet])
                })
                let mappedObj = {
                    count: listings?.length,
                    listings
                }
                res.status(200).json(mappedObj)
            } catch (e) {
                console.log(e)
                res.status(405).end(`Error fetching wallets`);
            }
            break;
        default:
            res.status(405).end(`Method ${method} Not Allowed`);
            break;
    }
}

export const config = {
    api: {
        externalResolver: true
    },
}

export default withSentry(handler)
```

And our connector:

```
import { initializeConnector } from "@web3-react/core"
import { WalletConnect } from "@web3-react/walletconnect"

import { URLS } from "../chains"

export const [walletConnect, hooks] = initializeConnector<WalletConnect>(
    (actions) =>
        new WalletConnect(actions, {
            rpc: URLS,
            qrcode: true,
            bridge: "https://bridge.walletconnect.org",
            qrcodeModalOptions: {
                registryUrl: "/api/walletconnect-registry",
                mobileLinks: [
                    "rainbow",
                ],
            },
        }),
    Object.keys(URLS).map((chainId) => Number(chainId))
)
```

**Comment by imthatcarlos** [comment]
nah you're legend for sharing the code 🙏 thanks @davidthomasparks , we're using next too so i'll plug this bad boy right in. cheers!

**Comment by dlazares** [comment]
any official fix coming?

**Comment by bkrem** [comment]
Hi everyone,

Is there a consistent/straightforward way to reproduce this issue? Spent some time looking into this, and so far I haven't found a way to reproduce this and I haven't seen it in the wild, neither with the `/api/v2/wallets` nor the legacy `wallets.json` endpoint. 

Additionally it's not clear to me why using the legacy endpoint as suggested here would change anything since they are now both handled by the identical Vercel infrastructure under the hood.

**Comment by davidthomasparks** [comment]
Hi @bkrem, it looks like the v2 endpoint is now properly fetching. Before we were getting 500 errors since it wasn't secured by SSL (or some other reason) but I just checked it and I can now see it's returning the wallet registy.

**Comment by bkrem** [comment]
Ok thanks for the update @davidthomasparks, great to hear this has resolved itself 🙌 

If this reoccurs at any point and/or has a reproduction case, please let us know here so we can reopen and investigate.
